spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -315,26 +306,26 @@ public OutputStream getOutputStream() throws IOException {
 	}
 
 	public String getBlobName() {
-		return this.blobName;
+		return this.location.getBlobName();
 	}
 
 	public String getBucketName() {
-		return this.bucketName;
+		return this.location.getBucketName();
 	}
 
 	/**
 	 * Check if this resource references a bucket and not a blob.
 	 * @return if the resource is bucket
 	 */
 	public boolean isBucket() {
-		return this.blobName == null;
+		return this.location.getBlobName() == null;
```

Comment:
Can you just move the whole method `isBucket` to the location class?

Reply:
Done.